// @nxtend/capacitor/src/schematics/capacitor-project/lib/add-project.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addProject = void 0;
const workspace_1 = require("@nrwl/workspace");
function addProject(options) {
    return workspace_1.updateWorkspaceInTree((json) => {
        const architect = json.projects[options.project].architect;
        const commands = ['add', 'copy', 'open', 'sync', 'update'];
        const platforms = ['ios', 'android'];
        architect['cap'] = {
            builder: '@nxtend/capacitor:cap',
            options: {
                cmd: '--help',
            },
        };
        let command, platform;
        for (command of commands) {
            const packageInstall = command === 'add' || command === 'update' || command === 'sync';
            architect[command] = {
                builder: `@nxtend/capacitor:cap`,
                options: {
                    cmd: `${command}`,
                    packageInstall,
                },
                configurations: {},
            };
            for (platform of platforms) {
                architect[command].configurations[platform] = {
                    cmd: `${command} ${platform}`,
                };
            }
        }
        return json;
    });
}
exports.addProject = addProject;
